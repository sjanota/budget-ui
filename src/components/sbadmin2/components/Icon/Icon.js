import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

export default function Icon({ icon, className, ...props }) {
  const classNames = classnames('fas', `fa-${icon}`, 'fa-fw', className);
  return <i className={classNames} {...props} />;
}

Icon.propTypes = {
  icon: PropTypes.string.isRequired,
  className: PropTypes.string,
};

Icon.Archive = 'archive';
Icon.Trash = 'trash-alt';
Icon.Edit = 'edit';
Icon.Plus = 'plus';
Icon.Minus = 'minus';
Icon.Save = 'save';
Icon.ClipboardCheck = 'clipboard-check';
Icon.ChevronUp = 'chevron-up';
Icon.ChevronDown = 'chevron-down';
