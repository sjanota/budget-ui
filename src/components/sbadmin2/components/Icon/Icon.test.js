import React from 'react';
import renderer from 'react-test-renderer';
import Icon from './Icon';

it('Icon renders with minimal props', () => {
  const component = renderer.create(<Icon icon={Icon.Archive} />);
  expect(component).toMatchSnapshot();
});
