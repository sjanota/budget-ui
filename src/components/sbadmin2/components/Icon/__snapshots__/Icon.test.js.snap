// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders with minimal props 1`] = `
<i
  className="fas fa-archive fa-fw"
/>
`;
